packages/cubejs-client-core/src/HttpTransport.js

```diff
@@ -2,34 +2,40 @@ import fetch from 'cross-fetch';
 import 'url-search-params-polyfill';
 
 class HttpTransport {
-  constructor({ authorization, apiUrl, headers = {}, credentials }) {
+  constructor({ authorization, apiUrl, method = 'GET', headers = {}, credentials }) {
```

Comment:
@ovr Should we also be changing the default to `POST` too?

Reply:
We should prefer the `GET` method whenever possible

Reply:
100% agree with @vasilev-alex, it's better to use GET whenever possible.

Reply:
@vasilev-alex @hassankhan Maybe it's better to set up it to `undefined` by default and detects automatically. 

```
method = method || (url.length < 2000 ? 'GET' : 'POST')
```

Reply:
@ovr yep, let's do like this. @mnifakram can you please introduce this check?

Reply:
yes sure thing

Reply:
@ovr @vasilev-alex I've just updated it. Please kindly check it. thanks